File: GFG_ALGO/src/org/main/gfg/java/algo/mathematics/Factorial.java
package org.main.gfg.java.algo.mathematics;

import java.util.Scanner;

public class Factorial {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int number = scan.nextInt();
		System.out.println(factorial(number, 1));
		
		
		
	}
	
	static int factorial(int number, int acc) {
		if (number <= 1) return acc;
		return factorial(number - 1, number * acc);
	}
}

File: SparkPractice/src/org/main/gfg/scala/algo/mathematics/Factorial.scala
package org.main.gfg.scala.algo.mathematics

import java.util.Scanner;

object Factorial extends App {
  val scan = new Scanner(System.in)
  val num = scan.nextInt()
  
  def fact(num: Int, acc: Int): Int = {
	  if (num <= 1) acc
	  else fact(num - 1, num * acc)
  }
  
  println(fact(num, 1))
}